modules/phase_field/src/kernels/EpsilonModelKernel2ndGauss.C

```diff
@@ -0,0 +1,41 @@
+
+#include "EpsilonModelKernel2ndGauss.h"
+
+registerMooseObject("PhaseFieldApp", EpsilonModelKernel2ndGauss);
+
+InputParameters
+EpsilonModelKernel2ndGauss::validParams()
+{
+  InputParameters params = ADKernel::validParams();
+  params.addClassDescription("Implements the term:-L\\nabla \\cdot \\left( \\frac{\\partial "
+                             "m(\\theta, v)}{\\partial \\nabla\\eta_i} f_0 \\right)");
+
+  params.addParam<MaterialPropertyName>("f_name", "F", " Base name of the free energy function");
+
+  return params;
+}
+
+EpsilonModelKernel2ndGauss::EpsilonModelKernel2ndGauss(const InputParameters & parameters)
+  : ADKernel(parameters),
+    _L_AD(getADMaterialProperty<Real>("L_AD")),
+
+    _dmdx(getADMaterialProperty<Real>("dmdx")),
+    _dmdy(getADMaterialProperty<Real>("dmdy")),
+    _dmdz(getADMaterialProperty<Real>("dmdz")),
```

Comment:
Yeah, this in fact should be a single `MaterialProperty<RealGradient>` (or `RealVectorValue`)

Reply:
That was fixed. See updated files.